pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -194,8 +194,12 @@ func (c *Cluster) Start() error {
 	if err != nil {
 		return errors.Wrapf(err, "failed to get information about currently-running OSD Deployments in namespace %q", namespace)
 	}
+	osdsToSkipReconcile, err := c.getOSDsToSkipReconcile()
+	if err != nil {
+		logger.Warningf("failed to get osds to skip reconcile. %v", err)
+	}
 	logger.Debugf("%d of %d OSD Deployments need updated", updateQueue.Len(), deployments.Len())
-	updateConfig := c.newUpdateConfig(config, updateQueue, deployments)
+	updateConfig := c.newUpdateConfig(config, updateQueue, deployments, osdsToSkipReconcile)
```

Comment:
What happens when the label is added to an OSD while the OSD update loop is already running? Are we worried about that corner case?

Reply:
This is an advanced case enough that I'm not worried about the corner case. If the reconcile does update the OSD, the OSD could get scaled up, but they could scale it down again, or restart the debug pod again if needed.